docker/service/src/main/java/uk/nhs/adaptors/scr/fhirmappings/CompositionMapper.java

```diff
@@ -0,0 +1,87 @@
+package uk.nhs.adaptors.scr.fhirmappings;
+
+import static uk.nhs.adaptors.scr.utils.DateUtil.formatDate;
+import static uk.nhs.adaptors.scr.utils.ThrowFhirMappingError.throwFhirMappingErrorMissingFromPayload;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hl7.fhir.r4.model.Composition;
+
+import uk.nhs.adaptors.scr.exceptions.FhirMappingException;
+import uk.nhs.adaptors.scr.models.GpSummary;
+import uk.nhs.adaptors.scr.models.Presentation;
+
+public class CompositionMapper {
+    public GpSummary mapComposition(GpSummary gpSummary, Composition composition) throws FhirMappingException {
+        if (composition == null) {
+            throwFhirMappingErrorMissingFromPayload("Composition");
+        }
+        setCompositionRelatesToId(gpSummary, composition);
+        setCompositionId(gpSummary, composition);
+        setCompositionDate(gpSummary, composition);
+        setPresentations(gpSummary, composition);
+
+        return gpSummary;
+    }
+
+    private static void setCompositionRelatesToId(GpSummary gpSummary, Composition composition) throws FhirMappingException {
+        String value = "";
+
+        if (composition.hasRelatesTo()) {
+            if (composition.getRelatesToFirstRep().hasTargetIdentifier()) {
+                if (composition.getRelatesToFirstRep().getTargetIdentifier().hasValue()) {
+                    value = composition.getRelatesToFirstRep().getTargetIdentifier().getValue();
+                } else {
+                    throwFhirMappingErrorMissingFromPayload("Composition RelatesTo TargetIdentifier Value");
+                }
+            } else {
+                throwFhirMappingErrorMissingFromPayload("Composition RelatesTo TargetIdentifier");
+            }
+        }
+
+        gpSummary.setCompositionRelatesToId(value);
+    }
+
+    private static void setCompositionId(GpSummary gpSummary, Composition composition) throws FhirMappingException {
+        String value = "";
+
+        if (composition.hasIdentifier()) {
```

Comment:
[Optional] Alternative approach:
```
gpSummary.setCompositionId(Optional.of(composition)
    .filter(Composition::hasIdentifier)
    .map(Composition::getIdentifier)
    .filter(Identifier::hasValue)
    .map(Identifier::getValue)
    .orElse(StringUtils.EMPTY));
```

Reply:
I think this approach would be less readable?